scale-tests/deploy-dispatchers.py

```diff
@@ -236,20 +228,24 @@ def deploy_dispatchers(
     """
     Deploy the required number of dispatchers and store their information to a text file.
     """
-
-    with open(output_file, "w") as outfile:
-        dispatchers = []
-        for i in range(num_dispatchers):
-            dispatcher_settings = install_package("spark", service_name_base, i,
-                                                linux_user,
-                                                lambda x: 0,
-                                                None,
-                                                options,
-                                                quota_options)
-            dispatchers.append(dispatcher_settings)
-            outfile.write("{},{},{}\n".format(dispatcher_settings["service"]["name"],
-                                              dispatcher_settings["roles"]["drivers"],
-                                              dispatcher_settings["roles"]["executors"]))
+    def deploy_dispatcher(i):
+        return install_package('spark',
+                               service_name_base,
+                               i,
+                               linux_user,
+                               lambda x: 0,
+                               None,
+                               options,
+                               quota_options)
+
+    with ThreadPoolExecutor(max_workers=MAX_THREADPOOL_WORKERS) as executor:
+        dispatchers = list(executor.map(deploy_dispatcher, range(num_dispatchers)))
+
+    with open(output_file, 'w') as outfile:
+        for dispatcher in dispatchers:
```

Comment:
Should we sort the dispatchers? Their order will now be non-deterministic, correct?

Reply:
The generator returned by `executor.map()` yields dispatchers in the same order of the collection mapped over.